packages/salesforcedx-vscode-core/src/context/workspaceOrgType.ts

```diff
@@ -13,21 +14,36 @@ export enum OrgType {
   NonSourceTracked
 }
 
+function imposeSlightDelay(ms = 0) {
+  return new Promise(r => setTimeout(r, ms));
+}
+
 export async function getWorkspaceOrgType(): Promise<OrgType> {
   const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
-  const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+  if (isNullOrUndefined(defaultUsernameOrAlias)) {
+    const e = new Error();
+    e.name = 'NoDefaultusernameSet';
+    throw e;
+  }
 
-  if (defaultUsernameIsSet) {
-    const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias!);
-    const isScratchOrg = await OrgAuthInfo.isAScratchOrg(username).catch(err =>
-      telemetryService.sendError(err)
+  let username;
+  let counter = 0;
+  while (isNullOrUndefined(username) && counter < 30) {
```

Comment:
Retrying here is what allows us to get the correct info from the cli when creating scratch orgs.

Reply:
can we do better here? i am not familiar with the code here but i wonder if there is a way to receive a notification when cli sets the scratch org context.

Reply:
Unfortunately no, the cli is not a server that sends notifications we can subscribe to. It's more of a sync api request an we get the data that's currently available.